File: dongwoo/week5/medium.java
import java.util.*;

class Solution {
    int solution(int[][] land) {
        int answer = 0;

        for (int i = 1; i < land.length; i++) {
            for (int j = 0; j < 4; j++) {
                int temp = 0;
                for (int z = 0; z < 4; z ++) {
                    if (j != z) {
                        temp = Math.max(temp, land[i-1][z]);
                    }
                }
                land[i][j] += temp;
            }
        }

        for (int l : land[land.length-1]) {
            answer = Math.max(l, answer);
        }

        return answer;
    }
}

File: jimin/week5/medium.py
def solution(land):
    n = len(land)
    dp = [[0]*4 for _ in range(n)]

    for j in range(4):
        dp[0][j] = land[0][j]

    for i in range(1, n):
        for j in range(4):
            best = 0
            for k in range(4):
                if k == j:
                    continue
                if dp[i-1][k] > best:
                    best = dp[i-1][k]
            dp[i][j] = land[i][j] + best

    return max(dp[n-1])